provider/pkg/provider/util/util.go

```diff
@@ -26,14 +28,20 @@ import (
 const PULUMI_COMMAND_STDOUT = "PULUMI_COMMAND_STDOUT"
 const PULUMI_COMMAND_STDERR = "PULUMI_COMMAND_STDERR"
 
-func CopyOutput(ctx p.Context, r io.Reader, doneCh chan<- struct{}, severity diag.Severity) {
+func LogOutput(ctx p.Context, r io.Reader, doneCh chan<- struct{}, severity diag.Severity) {
 	defer close(doneCh)
 	scanner := bufio.NewScanner(r)
 	for scanner.Scan() {
 		ctx.LogStatus(severity, scanner.Text())
 	}
 }
 
+// NoopLogger is for testing. It reads from the provided reader until EOF, discarding the output, then closes the channel.
+func NoopLogger(r io.Reader, done chan struct{}) {
```

Comment:
It seems like we use this outside of testing.

Reply:
True, thanks